### DISCUSSION TITLE
PushItemsAsync bug ?

### DISCUSSION THREAD

**Discussion by henda79** [Azure/azure-mobile-apps#570] [Q&A]
I have noticed an issue when trying to push a single table, but I'm not sure if its a bug or intended,

At the moment I have a single record in the _errors table for table A, now if I try to push table B I get an error which seems odd because the pushing of table B worked fine. It returns the error due to the previous error of table A now stored in the __errors table.

It is this code here in the PushItemsAsync method. It just gets all records from the __errors table and reports the problem.

Surely it should just get errors from the table I am trying to push ?

```
            List<TableOperationError> errors = new();
            PushStatus batchStatus = batch.AbortReason ?? PushStatus.Complete;
            try
            {
                errors.AddRange(await batch.LoadErrorsAsync(cancellationToken).ConfigureAwait(false));
            }
            catch (Exception ex)
            {
                batch.OtherErrors.Add(new OfflineStoreException("Failed to read errors from the local store.", ex));
            }
            SendPushFinishedEvent(batchStatus != PushStatus.Complete);

            // If the push did not complete successfully, then throw a PushFailedException.
            if (batchStatus != PushStatus.Complete || batch.HasErrors(errors))
            {
                List<TableOperationError> unhandledErrors = errors.Where(error => !error.Handled).ToList();
                Exception innerException = batch.OtherErrors.Count > 0 ? new AggregateException(batch.OtherErrors) : null;
                throw new PushFailedException(new PushCompletionResult(unhandledErrors, batchStatus), innerException);
            }
```

**Comment by adrianhall** [comment]
The errors for the batch is created for each batch.  

I should have a set of methods for dealing with the errors table, but that doesn't exist.  How do folks want that to work?
